File: src/engineeringenglish.py
import sys
ws = set()
for inp in sys.stdin:
    line = inp.split()
    lst = []
    for word in line:
        if word.lower() in ws:
            lst.append(".")
        else:
            lst.append(word); ws.add(word.lower())
    s = str()
    for i in lst:
        s += i + " "
    print(s)

File: src/engineeringenglish.cpp
#include<bits/stdc++.h>
using namespace std;

int main(){
    unordered_set<string> ws; string inp;
    while(getline(cin, inp)){
        istringstream iss(inp);
        string word, res;
        while(iss >> word){
            string lower;
            for(char& c: word){lower += tolower(c);}
            if(ws.find(lower) != ws.end()){res += ". ";}
            else{
                res += word + " ";
                ws.insert(lower);
            }
        }
        cout << res << endl;
    }
    return 0;
}